src/features/admission_plugins.rs

```diff
@@ -0,0 +1,207 @@
+use crate::{
+    cluster_api::{
+        clusterclasses::{
+            ClusterClassPatches, ClusterClassPatchesDefinitions,
+            ClusterClassPatchesDefinitionsJsonPatches,
+            ClusterClassPatchesDefinitionsJsonPatchesValueFrom,
+            ClusterClassPatchesDefinitionsSelector,
+            ClusterClassPatchesDefinitionsSelectorMatchResources, ClusterClassVariables,
+            ClusterClassVariablesSchema,
+        },
+        kubeadmcontrolplanetemplates::KubeadmControlPlaneTemplate,
+    },
+    features::{
+        ClusterClassVariablesSchemaExt, ClusterFeatureEntry, ClusterFeaturePatches,
+        ClusterFeatureVariables,
+    },
+};
+use cluster_feature_derive::ClusterFeatureValues;
+use kube::CustomResourceExt;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, ClusterFeatureValues)]
+pub struct FeatureValues {
+    #[serde(rename = "admissionControlList")]
+    pub admission_control_list: String,
+}
+
+pub struct Feature {}
+
+impl ClusterFeaturePatches for Feature {
+    fn patches(&self) -> Vec<ClusterClassPatches> {
+        vec![ClusterClassPatches {
+            name: "admissionControlList".into(),
+            enabled_if: Some("{{ if .admissionControlList }}true{{end}}".into()),
+            definitions: Some(vec![ClusterClassPatchesDefinitions {
+                selector: ClusterClassPatchesDefinitionsSelector {
+                    api_version: KubeadmControlPlaneTemplate::api_resource().api_version,
+                    kind: KubeadmControlPlaneTemplate::api_resource().kind,
+                    match_resources: ClusterClassPatchesDefinitionsSelectorMatchResources {
+                        control_plane: Some(true),
+                        ..Default::default()
+                    },
+                },
+                json_patches: vec![ClusterClassPatchesDefinitionsJsonPatches {
+                    op: "add".into(),
+                    path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/enable-admission-plugins".into(),
+                    value_from: Some(ClusterClassPatchesDefinitionsJsonPatchesValueFrom {
+                        variable: Some("admissionControlList".into()),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                }],
+            }]),
+            ..Default::default()
+        }]
+    }
+}
+
+inventory::submit! {
+    ClusterFeatureEntry{ feature: &Feature {} }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{features::test::TestClusterResources, resources::fixtures::default_values};
+    use maplit::btreemap;
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn test_disabled_if_admission_control_list_is_empty() {
+        let feature = Feature {};
+        let mut values = default_values();
+        values.admission_control_list = "".to_string();
```

Comment:
I think it will never be empty by the code logic.

Reply:
yeah, @klindgren-godaddy this will fail because of https://github.com/vexxhost/magnum-cluster-api/pull/751/files#diff-b21e2a96624fabac44d6c5ecc34bff0835565999be6a320fe75bd0401d5c8e87R430-R432